lib/build.js

```diff
@@ -249,21 +278,34 @@ class BuildModel extends BaseModel {
         const abortSteps = () => {
             const now = (new Date()).toISOString();
 
-            // Fail any running steps
-            this.steps = this.steps.map((step) => {
-                if (step.startTime && !step.endTime) {
-                    step.endTime = now;
-                    step.code = ABORT_CODE;
+            return this.getStepsModel().then((steps) => {
+                if (steps.length !== 0) {
+                    return Promise.all(steps.map((step) => {
+                        if (step.startTime && !step.endTime) {
+                            step.endTime = now;
+                            step.code = ABORT_CODE;
+                        }
+
+                        return step.update();
+                    }));
                 }
 
-                return step;
+                this.steps = this.steps.map((step) => {
```

Comment:
which case is this ? For older builds ? Does previous `this.getStepsModel()` need to be wrapped in `try-catch` ?

Reply:
This is for backward compatibility. `getStepsModel` is inside `abortSteps`.  The whole `abortSteps` function is in the promise chain.